plugins/module_utils/dcim.py

```diff
@@ -133,6 +141,20 @@ def run(self):
                 termination_b_name,
             )
 
+        elif endpoint_name == "module":
+            module_type_name = self.module.params["module_type"]
+            if self.module.params["parent_module_bay"]:
+                parent_name = self.module.params["parent_module_bay"].get("name")
+                if self.module.params["parent_module_bay"].get("parent_device"):
+                    parent_parent_name = self.module.params["parent_module_bay"].get("parent_device")
+                elif self.module.params["parent_module_bay"].get("parent_module"):
+                    parent_parent_name = self.module.params["parent_module_bay"].get("parent_module")
+            elif self.module.params["location"]:
+                parent_name = self.module.params["location"].get("name")
+                parent_parent_name = self.module.params["location"].get("parent", "—")
```

Comment:
Open to suggestions on this one.  I think other endpoints in the package are subject to the same failure unfortunately.  If I write logic to handle UUID, then I also have to fetch that UUID and parse out the pieces to make a unique name.  That's not really a pattern that currently exists from what I seen.  (I could have overlooked it somewhere though)

Reply:
Fair enough. At the very least please handle `parent_parent_name` and `parent_name` not being set due to the various `.get()` calls returning None.

Reply:
Added additional logic, error handling, and appropriate test case.

Reply:
I think this example will raise an exception since it is a dict and has a name, but no `parent_device` or `parent_module`

```
parent_module_bay:
  name: foo
```

We should probably just set `parent_name` and `parent_parent_name` at the top like this:

```
elif endpoint_name == "module":
    parent_name = None
    parent_parent_name = None
    <...>
```

I also think we shouldn't raise an error if they send the UUID. The main reason we use `raw` type is so they can pass the ID or a dict of attributes. We don't want to punish them for using the ID.

Reply:
so you want the name to be set to `<uuid> > <module_type>` if they pass a UUID?

Reply:
Sure, just set it to something rather than trying to use a variable that isn't defined.

Reply:
The name is really only used for the logs anyway

Reply:
It will always have some sort of name now.